### PR TITLE
Feature/ir 99/auth storage

### PR DESCRIPTION
**Pull request by gokceyetiserp**
Added sqlite db for storing login credentials.

### CODE REVIEW

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 1]
```diff
@@ -1268,19 +1272,98 @@ typedef struct {
...
+	int rc = sqlite3_open("credentials.db", &db);
+	if (rc) {
+		fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(db));
+		return (auth_record_t){0}; //return empty
+	}
+	else {
+		fprintf(stdout, "Opened database successfully\n");
+	}
+
+	createTable(db);
```
Comment (konstaty) [inline src/sp/protocol/mqtt/mqtt_parser.c:1331]: Why do you create DB every time you query for a auth record? (rhetorical question...)

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 2]
```diff
@@ -1268,19 +1272,98 @@ typedef struct {
...
+        auth_record.username = cparam->username.body;
+        auth_record.password = cparam->password.body;
+		insertCredentials(db, auth_record.clientid, auth_record.auth_type, auth_record.username, auth_record.password);
+    } 
+	else if (strstr(clientid, "client_jwt") == 0) {
+        auth_record.username = cparam->username.body;
+        auth_record.password = cparam->password.body;
+		if (auth_record.auth_type == MQTT_AUTH_JWT_ES256) {
+			auth_record.auth_type = MQTT_AUTH_JWT_ES256;
+			insertCredentials(db, auth_record.clientid, auth_record.auth_type, auth_record.username, auth_record.password);
```
Comment (konstaty) [inline src/sp/protocol/mqtt/mqtt_parser.c:1349]: The function ise called "**get**_auth_record" not **insert**_auth_method

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 3]
```diff
@@ -1268,19 +1272,98 @@ typedef struct {
...
+	}
+
+	createTable(db);
+
     auth_record_t auth_record;
     auth_record.clientid = clientid;
     auth_record.auth_type = MQTT_AUTH_INVALID;
+	//how to get auth_type?
+
     if (strcmp(clientid, "client_pas") == 0) {
```
Comment (konstaty) [inline src/sp/protocol/mqtt/mqtt_parser.c:1338]: The whole if/else block checking for a specific client ids is not needed anymore, please delete it. It was used as a temporal stub while we didn't have a supprort for DB.

**Review by konstaty — CHANGES_REQUESTED** [review]

File: src/sp/protocol/mqtt/mqtt_parser.c (hunk 1 above)
Comment (gokceyetiserp) [reply src/sp/protocol/mqtt/mqtt_parser.c:1331]: createTable function checks if db exists or not before creating, but i will change function.

File: src/sp/protocol/mqtt/mqtt_parser.c (hunk 2 above)
Comment (gokceyetiserp) [reply src/sp/protocol/mqtt/mqtt_parser.c:1349]: Oh, I am sorry, I thought we will get credentials from client. I fixed it.

**Pull request closed without merging**